src/cli-util.js

```diff
@@ -404,6 +410,77 @@ function createOptionUsageType(option) {
   }
 }
 
+function createDetailedUsage(optionName) {
+  const options = getOptionsWithOpposites(constant.detailedOptions);
+  const optionNames = options.map(option => option.name);
+
+  let optionIndex = optionNames.indexOf(optionName);
+
+  if (optionIndex === -1) {
+    const suggestionOptionIndex = optionNames.findIndex(
+      currentOptionName => leven(currentOptionName, optionName) < 3
+    );
+
+    if (suggestionOptionIndex !== -1) {
+      const suggestionOptionName = options[suggestionOptionIndex].name;
+      console.warn(
+        `Unexpected option name "${optionName}", did you mean "${suggestionOptionName}"?\n`
+      );
+      optionName = suggestionOptionName;
+      optionIndex = suggestionOptionIndex;
+    } else {
+      throw new Error(`Unexpected option name "${optionName}"`);
+    }
+  }
+
+  const option = options[optionIndex];
+
+  const optionTitleName = kebabToTitle(optionName);
+  const optionCamelName = kebabToCamel(optionName);
+
+  const header = `${optionTitleName} (${createOptionUsageHeader(option)})`;
+
+  const description = `\n\n${indent(option.description, 2)}`;
+
+  const choices =
+    option.type !== "choice"
+      ? ""
+      : (() => {
+          const choices = option.choices.filter(choice => !choice.deprecated);
+          const threshold =
+            choices
+              .map(choice => choice.value.length)
+              .reduce((current, length) => Math.max(current, length), 0) + 3;
+          const choiceUsages = choices.map(choice =>
+            indent(
+              createOptionUsageRow(choice.value, choice.description, threshold),
+              2
+            )
+          );
+          return `\n\nValid options:\n\n${choiceUsages.join("\n")}`;
+        })();
+
+  const defaults =
+    option.default !== undefined
+      ? `\n\nDefault: ${option.default}`
+      : optionCamelName in apiDefaultOptions
+        ? `\n\nDefault: ${apiDefaultOptions[optionCamelName]}`
+        : "";
+
+  return `${header}${description}${choices}${defaults}`;
+}
+
+function kebabToCamel(str) {
+  return str.replace(/-([a-z])/g, (_, char) => char.toUpperCase());
+}
+
+function kebabToTitle(str) {
+  return str.replace(
+    /(^|-)([a-z])/g,
+    (_, prefix, char) => " ".repeat(prefix.length) + char.toUpperCase()
```

Comment:
The `.repeat()` thing is almost a bit too clever to me :)

Do we even need `kebabToTile` though? Can't we just show `--bracket-spacing` instead of `Bracket Spacing` in the header?